Upgrade Spring Boot and Java

domonkos-ess opened the pull request:
Changes:

- Upgrade Spring Boot version to 3.4.4
- Upgrade Java version to 21
- Upgrade Jackson version to 2.18.3
- Upgrade Elasticsearch version to 8.18.0
- Align Docker related files
- Refactor WebSecurityConfig accordingly

Commit:
Uplift Spring Boot and Java versions

Upgrade the Spring Boot version to 3.x.x and the Java version to 21. Update dependencies for compatibility, and align the code with these upgrades, especially in WebSecurityConfig.

Commit:
Update Docker related files

Update Docker related files to use Java 21 and elasticsearch 8.18.0

Commit:
Update readme

Update readme to reference the correct versions.

Commit:
Update Github workflow files

Uplift Java version in teh Github workflow files

georgweiss reviewed (commented):
For the Olog service we're about to switch to:
* Java 25 (to benefit from language features like lightweight threads)
* Spring Boot 3.5.7
* Jackson 3.0.1
Would this be viable for Channel Finder aswell?

anderslindho commented:
@georgweiss we specified these versions for now, but we can create follow up for java 25 + highest spring boot 3, and then follow that with spring boot 4

anderslindho commented on README.md:
```diff
@@ -98,7 +98,7 @@ and [Eclipse](https://eclipseide.org/).
-    * JDK 17
+    * JDK 21
     * Maven (via package manager or via the wrapper `./mvnw`) (version specified
       in [the wrapper properties](./.mvn/wrapper/maven-wrapper.properties))
```
Don't you need to update also this? (maven wrapper)

domonkos-ess replied:
Should I upgrade it to 3.9+?

anderslindho replied:
Let's keep it out of scope - please create a backlog task re maven version

anderslindho reviewed (dismissed):
LGTM

@jacomago do we wanna set a requirement for the sonar lints - whether they must be addressed?

and do we want to do something about the tests?

georgweiss commented:
@anderslindho, since the Channel Finder service is an integral part of the Phoebus stack, it might be seen as a bit confusing to not run on the same base dependencies.
What is a "follow up" here in terms of planing and time schedule?

anderslindho commented:
@georgweiss

We do not have a precise time plan, as we are agile.

We can bump this in prio based on value, if you think this can cause confusion for maintainers. It would, however, be categorised as a maintenance change, to be prioritised lower than support and dev. Regardless, we often manage to squeeze in smaller maintenance updates in between support and dev work.

Do you have any targets for the rest of the services?

jacomago commented on compose.yml:
```diff
@@ -15,7 +15,7 @@ services:
 
   elasticsearch:
-    image: docker.elastic.co/elasticsearch/elasticsearch:8.11.2
+    image: docker.elastic.co/elasticsearch/elasticsearch:8.18.0
```
Can we update elastic in a different MR? It changes the requirements for running.

jacomago commented on src/main/java/org/phoebus/channelfinder/configuration/WebSecurityConfig.java:
```diff
@@ -83,71 +101,113 @@ public void configure(WebSecurity web) throws Exception {
     return new BCryptPasswordEncoder();
   }
+
+  private static class EmbeddedLdapCondition implements Condition {
```
This seems mixed up with demo auth and embedded ldap. They should be separate beans.

jacomago replied:
Can you rename this to DemoAuthCondition

jacomago commented on src/test/java/org/phoebus/channelfinder/processors/ChannelProcessorControllerIT.java:
```diff
@@ -30,7 +30,7 @@
 
   protected static final String AUTHORIZATION =
-      "Basic " + Base64Utils.encodeToString("admin:adminPass".getBytes());
+      "Basic " + Base64.getEncoder().encodeToString("admin:adminPass".getBytes());
```
Getting failures on this test. I'm guessing because of the mix of embeddedldap and demo auth.

georgweiss commented:
@anderslindho, Olog will be merged this week. For the other services I expect these to be migrated early next year.

Commit:
Undo elaticsearch upgrade

It have to be done later in a separate pull request.

Commit:
Correct demo auth config

Commit:
Resolve Sonar issues

Commit:
Resolve TaskExecutor bean conflict

Spring Boot 3 resolves TaskExecutor beans more strictly. Explicit bean naming avoids ambiguity with TaskScheduler.

Commit:
Remove trailing / from test andpoint

Commit:
Fix MetricsServiceIT REST calls

anderslindho commented:
Did you check on the sonar issue @domonkos-ess ?

Commit:
Document CSRF disbaled

CSRF disabled: application is a stateless REST API using HTTP Basic auth. No session or cookie-based authentication is used, so CSRF attacks are not applicable.

Commit:
Rename DemoAuthCondition

domonkos-ess commented:
> Did you check on the sonar issue @domonkos-ess ?

Yes, I saw it, however this is the equivalent of the previous implementation. SonarQube just didn’t take that into account. I documented the code.
